### ISSUE TITLE
Release 0.19.6

### ISSUE THREAD

**Issue by TobiGr** [TeamNewPipe/NewPipe#3862]
# Changelog
### New

- Add search for subscription picker in the feed group dialog #3373 
- Add filter to the feed group dialog to show only ungrouped subscriptions #3404 
- Add playlist tab to main page #3506
- Fast forward/rewind in background/pop-up player queue #3437 
- Display search suggestion: did you mean & showing result for #3471 
- Added translations for Arabic (Lybia), Bengali (India), Central Kurdish, Javanese and updated 31 others

### Improved

- Add Markdown export of crash logs #3579 
- Allow copying video title on long press #3772
- Do not remove failed streams from the queue  #3704
- Update status bar color to match toolbar color #3774
- Drop writing application metadata in muxed files #3843 
- Apply extractor changes: Add Page class and remove getNextPageUrl() #3441 
- [PeerTube] Add possibility to open links for some instances (without needing to share to NP) #3406, #3845 
- [Invidious]  Update list of instances #3841 

### Fixed

- Fixed audio/video desync caused by floating point cumulative errors #3837 
- Fixed crash when quickly swiping away search suggestions #3759
- Fixed enqueue resuming paused videos in popup player #3787 
- Fixed queue channel name color for some devices #3822 

### Development

-  Do not include Checkstyle in any APK #3828 
 
### New
-  Add Page class and remove getNextPageUrl() TeamNewPipe/NewPipeExtractor#314 

### Improved
- [PeerTube] Support embeded videos TeamNewPipe/NewPipeExtractor#344 
- [Invidious] Update instances TeamNewPipe/NewPipeExtractor#365 
- [YouTube] Use Integer operations in parseDurationString method TeamNewPipe/NewPipeExtractor#355 
- [YouTube] Add tests for parseDurationString() TeamNewPipe/NewPipeExtractor#338 
- [YouTube] Support `start=` timestamps TeamNewPipe/NewPipeExtractor#366 

### Fixed
- [SoundCloud] Fix concurrency issue when getting the client id TeamNewPipe/NewPipeExtractor#359 
- [PeerTube] Handle deleted comments TeamNewPipe/NewPipeExtractor#340 

### Development
- Use OkHttp for tests like in NewPipe TeamNewPipe/NewPipeExtractor#349 
- Expose extractor via API scope TeamNewPipe/NewPipeExtractor#361 

--------

# APK for testing
[NewPipe_v0.19.6_RC2.apk.zip](https://github.com/TeamNewPipe/NewPipe/files/4907559/NewPipe_v0.19.6_RC2.apk.zip)

**Comment by snappyapple632** [comment]
I found something involving importing settings. On a fresh installation (or after clearing all app data), the "What's New" page loads content from the "Trending" page instead of content from "What's New" after importing settings.

### Steps to reproduce
1. On a stable version of NewPipe, export your data and settings. Make sure in Settings>Content>Content of main page, "What's New" is selected.
2. Export your data on the stable NewPipe.
3. In RC1, use a fresh install or clear the storage in Android's settings.
4. Import your data on RC1. Tap "OK" on both prompts.
5. Go to the "What's New" tab. It will take a moment to load, then display content from the "Trending" tab until the app is restarted.

### Expected behavior
The "What's New" tab should display content from subscribed channels, regardless of the freshness of imported data.

### Actual behavior
The "What's New" tab displays content from the "Trending" tab when settings are imported for the first time.

**Comment by B0pol** [comment]
@snappyapple632 I know this issue is not specific to this release but I can't find the issue

**Comment by snappyapple632** [comment]
So it's been around for months? I hope we could get it fixed sometime later.

**Comment by opusforlife2** [comment]
What about the HLS/DASH videos bug fix? 🥺

**Comment by ondernuray** [comment]
Is there any possibility to add Pull #3556 Issue #2583 in this version? I would like to add my custom feeds also on main page and play all of them in the background with a single play button.

**Comment by Stypox** [comment]
@opusforlife2 @ondernuray you are pointing out important issues/prs, but those need to be reviewed thoroughly before merge and unfortunately this release was already postponed too much. Hopefully they will be in 0.20.0 :-D

**Comment by opusforlife2** [comment]
I'm aware it is unfinished, but the release cycle is a bit too slow currently, plus 0.20 is home to the gigantic unified player PR, not to mention tabs support, Bandcamp addition, notifications and so on... I don't see 0.20 being released any time before mid September, unless some PRs are postponed further. So I hope you'll release a hot fix just after the DASH/HLS PR is finished being reviewed and tested. A large class of YouTube videos face this issue, as you know.

**Comment by TobiGr** [comment]
@opusforlife2 We might want to ship a small update for that fix.

I suggest to release 0.19.6 tomorrow unless new bugs are reported.

**Comment by opusforlife2** [comment]
@TobiGr Thanks! That would be wonderful.

Also, a small observation: I've noticed in the 0.19 release series, especially, that old PRs tend to get sidelined by new PRs if they are small or medium sized features or bug fixes, which leads to postponement in the milestones. I'm no developer, but if a PR doesn't bring a high priority bug fix, wouldn't it be better to put it in a future milestone so that the PRs in the current milestone can be worked on consistently? It would free up PR authors to work on new things.

Of course, if the PR author isn't available for the release cycle in question, it makes sense to postpone the PR to the next milestone.

**Comment by wb9688** [comment]
![Screenshot of "Invalid format string"](https://user-images.githubusercontent.com/46277131/87230416-4bb47700-c3b0-11ea-8b84-43c484111c64.png)

![Screenshot of "Formatting argument types incomplete or inconsistent"](https://user-images.githubusercontent.com/46277131/87230438-71da1700-c3b0-11ea-9063-23f44809fbe5.png)

**Comment by TobiGr** [comment]
Fixed translations in second RC [NewPipe_v0.19.6_RC2.apk.zip](https://github.com/TeamNewPipe/NewPipe/files/4907559/NewPipe_v0.19.6_RC2.apk.zip)

**Comment by wb9688** [comment]
>What about the HLS/DASH videos bug fix? :pleading_face:

@opusforlife2: I personally don't think it's a good idea to do that until after #2907 has been merged, plus I'm not happy with the current implementation of that DASH "bug fix". I might do my own attempt soon on at least the NewPipeExtractor side though.

**Comment by opusforlife2** [comment]
@wb9688 I completely understand the intention to merge the best possible code. But this is code that exists and works _now_. It has worked for me perfectly so far and I use that debug apk whenever possible. Would it not suffice to use this code for now and improve the implementation in a later PR, whenever you find the time to work on it? You also have a lot of other PRs pending.

It's not like this is a corporate project where once something "just works" it will bit rot for years until resources need to be diverted to it again. You could take it up in the very next release cycle.

The reason I'm so vocal about this particular problem is that it gives the impression to newcomers and laypersons that Newpipe is still very buggy. It's important to get rid of that misconception ASAP. The other solution would be to temporarily have mp4 360p resolution selected by default, which is, again, not ideal at all.

Why ASAP? Because unlike many other FOSS apps which require explanations, comparisons and general convincing, Newpipe is an INSTANT WIN against the official app. You could meet someone in a shopping mall, have a five minute conversation about whatever, mention Newpipe (Download! Background playback! No ads!) and they readily agree to install it. But this bug, pretty quickly encountered, destroys all of that serendipity.

**Comment by wb9688** [comment]
@opusforlife2: I totally agree it's important, but some parts of that PR are unacceptable to me and it's better to do it right the first time, plus the NewPipe side doesn't support downloading for those streams. It's also just a waste of time to change anything about the NewPipe player until #2907 gets merged. I personally think v0.20.0 should be released as soon as #2907, #3178, #3777, and a proper fix for DASH gets merged, all other PRs would be a nice addition but not worth delaying v0.20.0 imho. #3178 and #3777 are necessary for proper Android 11 support.

**Comment by opusforlife2** [comment]
That gives me hope. I wish the rest of the team agrees with you.

**Comment by opusforlife2** [comment]
Did something change with respect to Soundcloud JSON response in 0.19.6? I'm getting a "Could not parse JSON response" error with a 0.19.5 apk, but 0.19.6 works fine.
